pkg/inject/sidecar.go

```diff
@@ -72,6 +72,11 @@ func Needed(dep *appsv1.Deployment, ns *corev1.Namespace) bool {
 		}).Debug("annotation not present, not injecting")
 		return false
 	}
+	// do not inject jaeger due to port collision
+	// do not inject if deployment's Annotation value is false
+	if dep.Labels["app"] == "jaeger" || dep.Annotations[Annotation] == "false" {
```

Comment:
Actually, I think `false` is already covered. If this value is set to anything other than `false` or an existing instance name, it will just skip the injection, as there won't be any suitable Jaeger instance:

https://github.com/jaegertracing/jaeger-operator/blob/16592e02c4089738e2156ce20349cb52037348cd/pkg/inject/sidecar_test.go#L186-L197

Reply:
The referenced test is correct. It does not inject the sidecar but there will be a log saying that Jaeger instance wasn't resolved for injection. 

We can pursue this direction, it seems more like a workaround solution rather than something what is supported.

Reply:
https://github.com/jaegertracing/jaeger-operator/issues/910